contracts/staking/PoSValidatorManager.sol

```diff
@@ -191,72 +192,73 @@ abstract contract PoSValidatorManager is IPoSValidatorManager, ValidatorManager
         PoSValidatorManagerStorage storage $ = _getPoSValidatorManagerStorage();
 
         // Unpack the Warp message
         WarpMessage memory warpMessage = _getPChainWarpMessage(messageIndex);
         (bytes32 validationID, uint64 nonce,) =
             ValidatorMessages.unpackSubnetValidatorWeightUpdateMessage(warpMessage.payload);
-        _checkPendingRegisterDelegatorMessages(validationID, delegator);
-        delete $._pendingRegisterDelegatorMessages[validationID][delegator];
+        _checkPendingRegisterDelegatorMessages(delegationID);
+        delete $._pendingRegisterDelegatorMessages[delegationID];
 
         Validator memory validator = _getValidator(validationID);
 
         // The received nonce should be no greater than the highest sent nonce
         require(validator.messageNonce >= nonce, "PoSValidatorManager: invalid nonce");
         // It should also be greater than or equal to the delegator's starting nonce
         require(
-            $._delegatorStakes[validationID][delegator].startingNonce <= nonce,
+            $._delegatorStakes[delegationID].startingNonce <= nonce,
             "PoSValidatorManager: nonce does not match"
         );
 
         require(
-            $._delegatorStakes[validationID][delegator].status == DelegatorStatus.PendingAdded,
+            $._delegatorStakes[delegationID].status == DelegatorStatus.PendingAdded,
             "PoSValidatorManager: delegator not pending added"
         );
         // Update the delegator status
-        $._delegatorStakes[validationID][delegator].status = DelegatorStatus.Active;
-        $._delegatorStakes[validationID][delegator].startedAt = uint64(block.timestamp);
-
+        $._delegatorStakes[delegationID].status = DelegatorStatus.Active;
+        $._delegatorStakes[delegationID].startedAt = uint64(block.timestamp);
         emit DelegatorRegistered({
             validationID: validationID,
-            delegator: delegator,
+            delegationID: delegationID,
             nonce: nonce,
             startTime: block.timestamp
         });
     }
 
     function initializeEndDelegation(
-        bytes32 validationID,
+        bytes32 delegationID,
         bool includeUptimeProof,
         uint32 messageIndex
     ) external {
+        PoSValidatorManagerStorage storage $ = _getPoSValidatorManagerStorage();
+        bytes32 validationID = $._delegatorStakes[delegationID].validationID;
+
         uint64 uptime;
         if (includeUptimeProof) {
             uptime = _getUptime(validationID, messageIndex);
         }
 
         // TODO: Calculate the delegator's reward, but do not unlock it
 
-        PoSValidatorManagerStorage storage $ = _getPoSValidatorManagerStorage();
-
         // Ensure the delegator is active
-        Delegator memory delegator = $._delegatorStakes[validationID][_msgSender()];
+        Delegator memory delegator = $._delegatorStakes[delegationID];
```

Comment:
Since we're no longer using `_msgSender()`, we need to explicitly check that this function is being called by the address that owns the delegation.